#include "chartmenu.h"
#include "ui_chartmenu.h"

chartMenu::chartMenu(QWidget *parent)
    : QMainWindow(parent)
    , ui(new Ui::chartMenu)
{
    ui->setupUi(this);

    connect(ui->barChartBtn, SIGNAL(clicked()), this, SLOT(createBarChart()));
    connect(ui->lineChartBtn, SIGNAL(clicked()), this, SLOT(createLineChart()));
    connect(ui->pieChartBtn, SIGNAL(clicked()), this, SLOT(createPieChart()));
}

chartMenu::~chartMenu()
{
    delete ui;
}

void chartMenu::createBarChart()
{
    QBarSet * set0 = new QBarSet("Jane");
    QBarSet * set1 = new QBarSet("John");
    QBarSet * set2 = new QBarSet("Axel");
    QBarSet * set3 = new QBarSet("Mary");
    QBarSet * set4 = new QBarSet("Samanta");

    *set0 << 1 << 2 << 3 << 4 << 5 << 6;
    *set1 << 5 << 0 << 0 << 4 << 0 << 7;
    *set2 << 3 << 5 << 8 << 13 << 8 << 5;
    *set3 << 5 << 6 << 7 << 3 << 4 << 5;
    *set4 << 9 << 7 << 5 << 3 << 1 << 2;

    QBarSeries *series = new QBarSeries();
    series->append(set0);
    series->append(set1);
    series->append(set2);
    series->append(set3);
    series->append(set4);

    QChart *chart = new QChart();
    chart->addSeries(series);
    chart->setTitle("Simple Barchart Example");
    chart->setAnimationOptions(QChart::SeriesAnimations);

    QStringList categories;
    categories << "Jan" << "Feb" << "Mar" << "Apr" << "May" << "Jun";

    QBarCategoryAxis *axisX = new QBarCategoryAxis();
    axisX->append(categories);
    chart->addAxis(axisX, Qt::AlignBottom);
    series->attachAxis(axisX);

    QValueAxis *axisY = new QValueAxis();
    axisY->setRange(0,15);
    chart->addAxis(axisY, Qt::AlignLeft);
    series->attachAxis(axisY);

    chart->legend()->setVisible(true);
    chart->legend()->setAlignment(Qt::AlignBottom);

    QChartView *chartView = new QChartView(chart);
    chartView->setRenderHint(QPainter::Antialiasing);

    QMainWindow *window = new QMainWindow(this);
    window->setCentralWidget(chartView);
    window->resize(420, 300);
    window->show();
}

void chartMenu::createLineChart()
{
    QLineSeries *series = new QLineSeries();

    series->append(0, 6);
    series->append(2, 4);
    series->append(3, 8);
    series->append(7, 4);
    series->append(10, 5);
    *series << QPointF(11, 1) << QPointF(13, 3) << QPointF(17, 6) << QPointF(18, 3) << QPointF(20, 2);

    QChart *chart = new QChart();
    chart->legend()->hide();
    chart->addSeries(series);
    chart->createDefaultAxes();
    chart->setTitle("Simple line chart example");

    QChartView *chartView = new QChartView(chart);
    chartView->setRenderHint(QPainter::Antialiasing);

    QMainWindow *window = new QMainWindow(this);
    window->setCentralWidget(chartView);
    window->resize(420, 300);
    window->show();
}

void chartMenu::createPieChart()
{
    QPieSeries *series = new QPieSeries();
    series->append("Jane", 1);
    series->append("Joe", 2);
    series->append("Andy", 3);
    series->append("Barbara", 4);
    series->append("Axel", 5);

    QPieSlice *slice = series->slices().at(1);
    slice->setExploded();
    slice->setLabelVisible();
    slice->setPen(QPen(Qt::darkGreen, 2));
    slice->setBrush(Qt::green);

    QChart *chart = new QChart();
    chart->addSeries(series);
    chart->setTitle("Simple Piechart Example");
    chart->legend()->hide();

    QChartView *chartView = new QChartView(chart);
    chartView->setRenderHint(QPainter::Antialiasing);

    QMainWindow *window = new QMainWindow(this);
    window->setCentralWidget(chartView);
    window->resize(420, 300);
    window->show();

}

